src/app/AllowCustomSchemesDialog.qml

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2019 Chris Clime
+ *
+ * This file is part of morph-browser.
+ *
+ * morph-browser is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * morph-browser is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import QtQuick 2.4
+import Ubuntu.Components 1.3
+
+ModalDialog {
+    objectName: "confirmDialog"
+    title: i18n.tr("Custom URL schemes")
+
+    property string url
+    property string domain
+    property bool showAllowPermanentlyCheckBox
+
+    message: i18n.tr("The site '%1' is trying to open the following URL with a custom scheme:").arg(domain) + "\n" +
+             url + "\n\n" +
+             i18n.tr("This will try to dispatch this url to a system app if possible. Do you want to proceed?");
+
+    signal allow()
+    signal allowPermanently()
+    signal cancel()
+    
+    onAllow: hide()
+    onAllowPermanently: hide()
+    onCancel: hide()
+
+    ListItemLayout {
+        visible: showAllowPermanentlyCheckBox
+        title.text: i18n.tr("Remember decision")
+        CheckBox {
+            id: allowPermanentlyCheckBox
+         }
+    }
+    Button {
+        text: i18n.tr("Allow")
```

Comment:
I noticed that allowing or denying one of these is remembered until the end of the session. That is, if I deny a URL open without checking "Remember decision" then click the URL again, I'm not asked to perform the action again.

Now, this is good for not spamming the user. Is it what you expected to happen?

Reply:
yes this is done on purpose. Since the for geolocation requests this behavior (remembering the decision for the current session) is already done by the backend (QtWebEngine), I wanted the same behavior here.

By the way: in incognito mode the "remember decision" checkboxes are not available.

Reply:
Okay, just making sure.